traffic_monitor/health/cache.go

```diff
@@ -175,65 +180,127 @@ func CalcAvailability(results []cache.Result, pollerName string, statResultHisto
+				localCacheStatuses[tc.CacheName(result.ID)] = make(map[string]cache.AvailableStatus)
 			}
-			log.Infof("Changing state for %s was: %t now: %t because %s poller: %v on protocol %v error: %v", result.ID, available.IsAvailable, newAvailableState, whyAvailable, pollerName, protocol, result.Error)
-			events.Add(Event{Time: Time(time.Now()), Description: "Protocol: (" + protocol + ") " + whyAvailable + " (" + pollerName + ")", Name: string(result.ID), Hostname: string(result.ID), Type: toData.ServerTypes[tc.CacheName(result.ID)].String(), Available: newAvailableState, IPv4Available: availableTuple.IPv4, IPv6Available: availableTuple.IPv6})
-		}
 
-		localStates.SetCache(tc.CacheName(result.ID), tc.IsAvailable{IsAvailable: newAvailableState, Ipv4Available: availableTuple.IPv4, Ipv6Available: availableTuple.IPv6})
+			localCacheStatuses[tc.CacheName(result.ID)][interfaceName] = cache.AvailableStatus{
+				Available:          availableTuple,
+				ProcessedAvailable: newAvailableState,
+				Status:             mc.TrafficServer[string(result.ID)].ServerStatus,
+				Why:                whyAvailable,
+				UnavailableStat:    unavailableStat,
+				Poller:             pollerName,
+				LastCheckedIPv4:    usingIPv4,
+			} // TODO move within localStates?
+
+			if available, ok := localStates.GetCache(tc.CacheName(result.ID)); !ok || available.IsAvailable != newAvailableState {
+				protocol := "IPv4"
+				if !usingIPv4 {
+					protocol = "IPv6"
+				}
+				log.Infof("Changing state for %s was: %t now: %t because %s poller: %v on protocol %v error: %v", result.ID, available.IsAvailable, newAvailableState, whyAvailable, pollerName, protocol, result.Error)
+				events.Add(Event{Time: Time(time.Now()), Description: "Protocol: (" + protocol + ") " + whyAvailable + " (" + pollerName + ")", Name: string(result.ID), Hostname: string(result.ID), Type: toData.ServerTypes[tc.CacheName(result.ID)].String(), Available: newAvailableState, IPv4Available: availableTuple.IPv4, IPv6Available: availableTuple.IPv6})
+			}
+
+			// Compute aggregate data based on each interface
+			aggregateStatus := cache.AvailableStatus{
+				Available: cache.AvailableTuple{
+					IPv4: false,
+					IPv6: false,
+				},
+				ProcessedAvailable: false,
+				LastCheckedIPv4:    false,
+				Status:             "",
+				Why:                "",
+				UnavailableStat:    "",
+				Poller:             "",
+			}
+			for interfaceName, status := range localCacheStatuses[tc.CacheName(result.ID)] {
+				if interfaceName == tc.CacheInterfacesAggregate {
+					continue
+				}
+				aggregateStatus.Available.IPv4 = aggregateStatus.Available.IPv4 || status.Available.IPv4
+				aggregateStatus.Available.IPv6 = aggregateStatus.Available.IPv6 || status.Available.IPv6
+
+				// What does this mean on aggregated data?
+				// For now assume that if any interface was then the aggregate is
+				aggregateStatus.LastCheckedIPv4 = aggregateStatus.Available.IPv4 || status.LastCheckedIPv4
```

Comment:
maybe copy leftover? should be `aggregateStatus.LastCheckedIPv4 = aggregateStatus.LastCheckedIPv4 || status.LastCheckedIPv4` right?

Reply:
Yep, good catch